### PR TITLE
Make all ChefDK components top-level dependencies

### PR DESCRIPTION
**Pull request by tyler-ball**
Instead, package them as real Omnibus Software definitions. 

Related PR: https://github.com/chef/omnibus-software/pull/519

Fixes: https://github.com/chef/chef-dk/issues/593

\cc @schisamo

### CODE REVIEW

**Comment by tyler-ball** [comment]
Tagging @mwrock because he had a comment about not knowing which version of winrm-transport we pin in the ChefDK.

This won't prevent something else from pulling in winrm-transport as a transitive dependency and getting a different version, but I don't know of anything that does that.

**Comment by schisamo** [comment]
Shouldn't we just push the following software defs up into omnibus-software (since you are already doing that for RuboCop):

```
config/software/chefspec.rb
config/software/fauxhai.rb
config/software/knife-spork.rb
config/software/knife-windows.rb
```

**Comment by tyler-ball** [comment]
We can - I didn't know the correct pattern to follow.  Because other chef workflow specific tools were in `omnibus-chef` (like inspec, chef-provisioning, chef-vault) I put these here.  I'm hoping you can just tell me the correct thing to do :)

File: config/projects/chefdk.rb [hunk 1]
```diff
@@ -67,9 +72,12 @@
 
 override :inspec, version: "v0.9.2"
 
+override :rubocop, version: "v0.31.0"
+
 override :'test-kitchen', version: "v1.4.2"
 override :'kitchen-inspec', version: "v0.9.0"
 override :'kitchen-vagrant', version: "v0.19.0"
+override :'winrm-transport', version: "v1.0.2"
```
Comment (mwrock) [inline config/projects/chefdk.rb]: just released v1.0.3 so lets update that

**Comment by schisamo** [comment]
@tyler-ball We should just push everything up into omnibus-software.

**Comment by schisamo** [comment]
@tyler-ball Looks like all of these were missed:

* chef-provisioning
* chef-provisioning-*
* chef-vault
* foodcritic
* inspec
* kitchen-inspec
* kitchen-vagrant

**Pull request merged**